Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/AbstractDocSubmissionAuditTransforms.java

```diff
@@ -0,0 +1,149 @@
+
+        ParticipantObjectIdentificationType participantObject = createParticipantObject(
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_ROLE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_DISPLAY_NAME);
+
+        if (NullChecker.isNotNullish(pid)) {
+            participantObject.setParticipantObjectID(pid);
+        }
+        return participantObject;
+    }
+
+    private ParticipantObjectIdentificationType createSubmissionSetParticipantObjectIdentification(String submissionId) {
+
+        ParticipantObjectIdentificationType participantObject = createParticipantObject(
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_TYPE_CODE_ROLE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_CODE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_DISPLAY_NAME);
+
+        if (NullChecker.isNotNullish(submissionId)) {
+            participantObject.setParticipantObjectID(submissionId);
+        }
+        return participantObject;
+    }
+
+    private ParticipantObjectIdentificationType createParticipantObject(short objTypeCodeSys, short objTypeCodeRole,
+        String objIdTypeCode, String objIdTypeCodeSys, String objIdTypeDisplayName) {
+
+        return createParticipantObjectIdentification(objTypeCodeSys, objTypeCodeRole,
+            objIdTypeCode, objIdTypeCodeSys, objIdTypeDisplayName);
+    }
+
+    private String getIdValue(ProvideAndRegisterDocumentSetRequestType request, String idType) {
+        String idValue = null;
+        RegistryObjectType registryObj = extractRegistryObject(
+            request.getSubmitObjectsRequest().getRegistryObjectList());
+        if (registryObj != null && registryObj.getExternalIdentifier() != null
+            && registryObj.getExternalIdentifier().size() > 0) {
+            idValue = getIdFromExternalIdentifiers(registryObj.getExternalIdentifier(), idType);
+        }
+        return idValue;
+    }
+
+    private String getIdFromExternalIdentifiers(List<ExternalIdentifierType> externalIdentifiers, String type) {
+        String id = null;
+        for (ExternalIdentifierType identifier : externalIdentifiers) {
+            if (identifier.getName() != null
+                && identifier.getName().getLocalizedString() != null
+                && !identifier.getName().getLocalizedString().isEmpty()
```

Comment:
Sorry! I have added comments on individual commits.
Adding it again here to track class and line number
I have seen NullChecker class used for check in QD and RD pr's Yesterday to check list values. To be consistent Can this be a NullChecker check for the list here?